### PR TITLE
add option to use width as percentage

### PR DESCRIPTION
**Pull request by booperlv**
adds option to use nvim_tree_width as a percentage (reads the width as a percentage and converts to decimal, multiply to vim total columns)

### CODE REVIEW

File: lua/nvim-tree/view.lua [hunk 1]
```diff
@@ -224,7 +224,16 @@ function M.open()
   a.nvim_command("vsp")
   local move_to = move_tbl[M.View.side]
   a.nvim_command("wincmd "..move_to)
-  a.nvim_command("vertical resize "..M.View.width)
+
+  local width
+  if (vim.g.nvim_tree_width_as_percent == true) then
```
Comment (kyazdani42) [inline lua/nvim-tree/view.lua:229]: no parens in lua conditions please :)

File: lua/nvim-tree/view.lua [hunk 2]
```diff
@@ -224,7 +224,16 @@ function M.open()
...
+    local percentAsDecimal = M.View.width/100
```
Comment (kyazdani42) [inline lua/nvim-tree/view.lua:230]: space around /. Also use snake_case

File: lua/nvim-tree/view.lua [hunk 3]
```diff
@@ -224,7 +224,16 @@ function M.open()
...
+    width = math.floor( vim.o.columns * percentAsDecimal )
```
Comment (kyazdani42) [inline lua/nvim-tree/view.lua:231]: remove spaces after opening parens and before closing parens

**Review by kyazdani42 — CHANGES_REQUESTED** [review]
thanks for the feature, a few formatting issues to fix before merging :)

File: lua/nvim-tree/view.lua (hunk 1 above)
Comment (gegoune) [reply lua/nvim-tree/view.lua:229]: Should that not be ` == 1`? Aren't all other option in nvim-tree `0/1` rather than `true/false`?

**Review by kyazdani42 — CHANGES_REQUESTED** [review]
oh yes and please add documentation for the option

File: lua/nvim-tree/view.lua (hunk 3 above)
Comment (gegoune) [reply lua/nvim-tree/view.lua:231]: Could just add some formatter to CI checking each PR for style issues. It's good to have consistent styling throughout the project.

**Comment by booperlv** [comment]
Sorry for the odd code, whipped this up as a quick fix while on a break from other projects.
Will clean up in a few, Thanks!

**Comment by booperlv** [comment]
One more question by the way, Where do you think it would be best to place the option in the documentation? Would right under g:nvim_tree_width be better or is it good as is right now?

**Comment by sindrets** [comment]
Some thoughts: do we really need another option for this? Seems really unnecessary. Can we not just make it so that `nvim_tree_width` also accepts a string, and then we can determine whether or not it's a percentage depending on the value.

```vim
let g:nvim_tree_width = 30 " indicates width of 30 columns.
let g:nvim_tree_width = '30%' " indicates width of 30% of '&columns'.
```

**Comment by gegoune** [comment]
That is very reasonable.

**Comment by kyazdani42** [comment]
agreed with sindrets proposal, looks better this way :)

**Comment by booperlv** [comment]
alright, going to start working on it :)

**Comment by booperlv** [comment]
Opened a new pull request ( #473 ) with the suggested changes, both because I think it's different, and because I messed up resetting the fork's master branch and ended up deleting the fork. The more I learn about git and not messing around with it at 2AM. Closing this one now :D

**Pull request closed without merging**